Deprecate `num_steps` in `qml.evolve` (#7954)
PennyLaneAI/pennylane: doc/development/deprecations.rst

Pending deprecations
--------------------

* Providing ``num_steps`` to ``qml.evolve`` and ``Evolution`` is deprecated and will be removed in a future version.

>   Instead, you can use :class:`~.TrotterProduct` providing the ``n`` parameter to perform the
>   Suzuki-Trotter product approximation of a Hamiltonian with the specified number of Trotter steps.
> 

Reviewer: ```suggestion
  Instead, use :class:`~.TrotterProduct` for approximate methods, providing the ``n`` parameter to perform the
  Suzuki-Trotter product approximation of a Hamiltonian with the specified number of Trotter steps.

```
